src/main/java/app/AppBuilder.java

```diff
@@ -116,11 +117,12 @@ public AppBuilder addSignupUseCase() {
      * Adds the Login Use Case to the application.
      * @return this builder
      */
-    public AppBuilder addLoginUseCase() {
+    public AppBuilder addLoginUseCase(LoginUserDataAccessInterface loginUserDAO) {
```

Comment:
You added a parameter to this app builder method, it seems like it still follows clean architecture but just make sure. If it doesn't another option could be to make the initialization withthin app builder.